components/renku_data_services/solr/solr_client.py

```diff
@@ -364,3 +413,71 @@ async def delete(self, query: str) -> Response:
     async def close(self) -> None:
         """Close this client and free resources."""
         return await self.delegate.aclose()
+
+
+class SolrAdminClient(AbstractAsyncContextManager, ABC):
+    """A client to the core admin api.
+
+    Url: https://solr.apache.org/guide/solr/latest/configuration-guide/coreadmin-api.html
+    """
+
+    @abstractmethod
+    async def status(self, core_name: str | None) -> dict[str, Any] | None:
+        """Return the status of the connected core."""
+        ...
+
+    async def create(self, core_name: str | None) -> None:
```

Comment:
question: This is not `@abstractmethod` on purpose?

Reply:
Ah no thanks, just forgot (not used to marking it like this… 😅)